services/movie_session.py

```diff
@@ -0,0 +1,52 @@
+from datetime import datetime
+from typing import Optional
+
+from django.db.models.query import QuerySet
+
+from db.models import MovieSession
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> None:
+    MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id
+    )
+
+
+def get_movies_sessions(session_date: Optional[str] = None) -> QuerySet:
+    movie_sessions = MovieSession.objects.all()
+    if session_date:
+        movie_sessions = movie_sessions.filter(show_time__date=session_date)
+    return movie_sessions
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: Optional[datetime] = None,
+        movie_id: Optional[int] = None,
+        cinema_hall_id: Optional[int] = None
+) -> None:
+    if show_time is None and movie_id is None and cinema_hall_id is None:
```

Comment:
Nice try but you can make it better. Try to follow these steps:
1. Create an empty dict
2. Fill out the dict one by one in your if statements in case if you received value to update
3. IF you have any values in dict in the end -> update session with update dict (no need to get a movie session in the beginning)